addon/decorators/track-on-constructor.ts

```diff
@@ -0,0 +1,29 @@
+import { ActivityType } from '../services/activity-tracking';
+import { assert } from '@ember/debug';
+import { getOwner } from '@ember/application';
+
+interface Constructable {
+  new (...args: any[]): any;
+}
+
+export function trackOnConstructor(actionDescription: string, actionType: ActivityType) {
+  return function extendClass<T extends Constructable>(BaseClass: T) {
+    return class extends BaseClass {
+      constructor(...args: any) {
+        super(...args);
+        assert(
+          '[decorator][track-on-constructor] An actionDescription needs to be passed for the activity-log to make sense.',
+          typeof actionDescription === 'string'
+        );
+        getOwner(this)
+          .lookup('service:activity-tracking')
+          .log({
+            type: actionType ?? 'page_view',
+            route: window.location.origin,
```

Comment:
```
location.origin

The origin read-only property of the [Location](dfile:///Users/alexis/Library/Application%20Support/Dash/DocSets/JavaScript/JavaScript.docset/Contents/Resources/Documents/developer.mozilla.org/en-US/docs/Web/API/Location.html) interface is a string containing the Unicode serialization of the origin of the represented URL.
```

i dont think we quite want it here, my goal here is to have a low cardianility unique identifier to specific a specific action.

For example i want to know who checks out the suggestion panel in the list: The path would be `/lists/1234` (w/ ?status=suggested) but also `/lists/2345`. I need something reusable, i.e.  `lists/show/suggested` or something independent from the data rendered.

Reply:
Hmmm, I think we'll need to discuss this live as well.

route & or path give us little information in the software since we have a big tendency to use modals (in modals, in modals...).

In a real use case:
Ember transitions will be tracked by calling the ActivityTracking service directly and then opening a feature we can use the decorator on the Ctor for a modal along with the `log-activity` helper on a button.
Which would, for instance, give us the user flow of : transition to X page, click on X button, open X modal, close X modal, transition to X, etc.